packages/core-ethereum/crates/core-ethereum-api/src/lib.rs

```diff
@@ -31,46 +31,46 @@ pub trait ChainQueries {
     async fn clone(&self) -> Self;
 }
 
-/// This is used by the indexer to emit events when a change on channel entry is detected.
-#[derive(Clone)]
-pub struct ChannelEventEmitter {
-    pub tx: UnboundedSender<ChannelEntry>,
-}
-
-impl ChannelEventEmitter {
-    pub async fn send_event(&self, channel: &ChannelEntry) {
-        let mut sender = self.tx.clone();
-        let _ = sender.send(channel.clone()).await;
-    }
-}
 
 #[cfg_attr(feature = "wasm", wasm_bindgen::prelude::wasm_bindgen)]
 #[derive(Clone)]
 pub struct HoprChain {
     me_onchain: ChainKeypair,
     db: Arc<RwLock<CoreEthereumDb<utils_db::rusty::RustyLevelDbShim>>>,
+    // indexer: Indexer<>,      // TODO: we need RPC at this point
     chain_actions: CoreEthereumActions<CoreEthereumDb<RustyLevelDbShim>>,
-    channel_events: ChannelEventEmitter,
     channel_graph: Arc<RwLock<core_path::channel_graph::ChannelGraph>>,
 }
 
 impl HoprChain {
     pub fn new(
         me_onchain: ChainKeypair,
         db: Arc<RwLock<CoreEthereumDb<utils_db::rusty::RustyLevelDbShim>>>,
+        // rpc...
```

Comment:
An object should be passed here, that can be used to register `on_change` operation for the indexer stream block processing. That object will be used by the core ethereum actions facade to identify whether the desired operation has been recorded on-chain by observing the changed state and not the tx hash.

Reply:
Resolved in https://github.com/hoprnet/hoprnet/pull/5749